### PR TITLE
ARC-290 Make logs more informative and make sure no sensitive data logged

### PR DESCRIPTION
**Pull request by KAbakumovAtl**
In this Pull Request I've cleaned up logging. Added some logs to make sure we log sufficient information, changed logger names.

I've also changed some HTTP responses, so we'll return 500 error codes on unknown errors (instead of 400)

### CODE REVIEW

File: src/config/enhance-octokit.ts [hunk 1]
```diff
@@ -31,7 +29,6 @@ const instrumentRequests = (octokit: GitHubAPI, log: LoggerWithTarget) => {
       };
 
       statsd.histogram(metricHttpRequest().github, elapsed, tags);
-      log.debug(tags, `GitHub request time: ${elapsed}ms`);
```
Comment (KAbakumovAtl) [inline src/config/enhance-octokit.ts:34]: Removed it to not spam console output when testing. We have a metric anyway.

File: src/config/statsd.ts [hunk 2]
```diff
@@ -11,14 +11,19 @@ export const globalTags = {
...
 
 const statsd = new StatsD({
   prefix: 'github-for-jira.',
   host: 'platform-statsd',
   port: 8125,
   globalTags,
-  errorHandler: (err) => logger.warn({ err }, 'error writing metrics'),
+  errorHandler: (err) => {
+    if (process.env.NODE_ENV !== 'development') {
+      logger.warn(err, 'error writing metrics')
```
Comment (KAbakumovAtl) [inline src/config/statsd.ts:23]: I've added a check to not log these errors when we test locally, hence there will be a lot

File: src/frontend/app.ts [hunk 3]
```diff
@@ -241,10 +241,9 @@ export default (octokitApp: App): Express => {
       'Not Found': 404,
     };
 
-    const errorStatusCode = errorCodes[err.message] || 400
+    const errorStatusCode = errorCodes[err.message] || 500
```
Comment (KAbakumovAtl) [inline src/frontend/app.ts:244]: If we were unable to resolve the error code by exception, we'll return 500 status code, hence the error is unknown. ( We can't blame the client if we not sure 🙂 )

File: src/frontend/delete-github-subscription.ts [hunk 4]
```diff
@@ -79,7 +82,7 @@ export default async (req: Request, res: Response): Promise<void> => {
         });
     }
   } catch (err) {
-    req.log.error(err);
-    res.sendStatus(400);
+    req.log.error(err, 'Error while processing delete subscription request');
+    res.sendStatus(500);
```
Comment (KAbakumovAtl) [inline src/frontend/delete-github-subscription.ts:86]: Same here, return 500 if we caught an unknown error

File: src/frontend/github-client-middleware.ts [hunk 5]
```diff
@@ -40,7 +39,7 @@ export const isAdmin = (githubClient: GitHubAPI) =>
       } = await githubClient.orgs.getMembership({ org, username });
       return role === "admin";
     } catch (err) {
-      logger.error(`${org} has not accepted new permission for getOrgMembership - error: ${err}`);
+      logger.warn(err, `${org} has not accepted new permission for getOrgMembership`);
```
Comment (KAbakumovAtl) [inline src/frontend/github-client-middleware.ts:42]: We'll pass req.log here, so we log it with the request Id

File: src/github/middleware.ts [hunk 6]
```diff
@@ -81,7 +81,6 @@ export default (
         {
           noop: 'bot',
           botId: context.payload.sender.id,
-          botLogin: context.payload.sender.login,
```
Comment (KAbakumovAtl) [inline src/github/middleware.ts:84]: I was unable to reproduce this log, however `botLogin` doesn't sound like something we can log

File: src/github/index.ts [hunk 7]
```diff
@@ -10,12 +10,15 @@ import { getLogger } from '../config/logger'
...
 export default (robot) => {
-  const logger = getLogger('github');
+  const logger = getLogger('github.webhooks');
 
   // TODO: Need ability to remove these listeners, especially for testing...
   robot.on('*', (context) => {
     const { name, payload } = context;
-    logger.info({ event: name, action: payload.action }, 'Event received');
+
+    context.log = logger.child({id: context.id})
```
Comment (KAbakumovAtl) [inline src/github/index.ts:19]: Later we should add a proper tracing in our app. We need to put this ID in our queue payloads, and use it to trace our logs. Same for Installation jobs, we need to have a trace ID there, so we can query all logs related to the same sync. But this is quite a big task and out of scope of this PR

**Review by rachellerathbone — APPROVED** [review]
Looks good to me :)

**Pull request merged**